crowbar_framework/app/models/provisioner_service.rb

```diff
@@ -177,5 +177,66 @@ def transition(inst, name, state)
     @logger.debug("Provisioner transition: exiting for #{name} for #{state}")
     [200, { name: name }]
   end
-end
 
+  def load_repository_bag
+    bag = Chef::DataBag.load("repositories") rescue nil
+    if bag.nil?
+      create_repository_bag
+    end
+    bag
+  end
+
+  def synchronize_repositories(platforms)
+    platforms.each do |platform, repos|
+      repos.each do |reponame, active|
+        repo_object = Crowbar::Repository.where(platform: platform, repo: reponame).first
+        case active
+        when "0"
+          db = Chef::DataBagItem.load("repositories", repo_object.id) rescue nil
+          # skip repo if it hasn't been active before
+          next if db.nil?
+          destroy_repository_item(db)
+        when "1"
+          create_repository_item(repo_object)
+        end
+      end
+    end
+  end
+
+  def create_repository_item(repo_object)
+    # create DataBag if it doesn't exist yet
+    load_repository_bag
+    # create DataBagItem with the repo metadata used by provisioner recipes
+    repository_item = Chef::DataBagItem.new
+    repository_item.data_bag "repositories"
+    repository_item["id"] = repo_object.id
```

Comment:
Maybe you should add a `.to_data_bag` method to the `repo_object` to simplify that?